### DISCUSSION TITLE
Undefined method

### DISCUSSION THREAD

**Discussion by AbdelilahDerfoufi** [symfony/symfony#45321] [Q&A]
I have this as a get method in my user entity  but when I call it, I get an error message, why and how can I fix it?

 $user = $this->security->getUser();
        $intervention = $user->getInterventions();
        $intervention;

        return $this->render('profil/index.html.twig');

**Comment by helmis123** [comment]
Is your method "Public" ?

**Reply by AbdelilahDerfoufi** [reply]
Yes.
![Screenshot_20220205-001551_Chrome](https://user-images.githubusercontent.com/49759161/152615588-e63f2c11-5016-4c67-ae05-0d77ad32a6d8.jpg)

**Reply by AbdelilahDerfoufi** [reply]
And I'm using visuel studio code.

**Reply by chyshyya** [reply]
Please provide:
1. symfony version
2. service or controller you try to get user
3. dd($user). Place it after you get you user from security class and give an informatrion if user are there at all or null

**Reply by AbdelilahDerfoufi** [reply]
1- Symfony version: 5.4
2- The user object is fetched but Interventions (a one to many relation) I can't get them within the user object.
3- ProfilController : I am trying to get every intervention of the current user.

**Reply by chyshyya** [reply]
Try simply `$user = $this->getUser();` in controller

**Reply by AbdelilahDerfoufi** [reply]
It works this one, it's the getInterventions method is what does not work, it's underlined with red and the error is undefined method.
The getter is not recognised.

**Reply by chyshyya** [reply]
Could you post files?
1. user entity
2. intervention entity
3. securyty.yaml. Remove everything really secure if needed.

**Reply by chyshyya** [reply]
I'll add to my project and test it for you
And post controller method where it doesn't work propertly.

**Reply by AbdelilahDerfoufi** [reply]
[Intervention.zip](https://github.com/symfony/symfony/files/8013701/Intervention.zip)
[User.zip](https://github.com/symfony/symfony/files/8013706/User.zip)
[security.zip](https://github.com/symfony/symfony/files/8013711/security.zip)

**Reply by chyshyya** [reply]
Also I need information for testing `php -v`

**Reply by AbdelilahDerfoufi** [reply]
Php 8.0.2
And I apologize for the trouble.

**Reply by norkunas** [reply]
Probably from the start would be better if you have posted full error stacktrace :)

**Reply by AbdelilahDerfoufi** [reply]
There is no error stacktrace that's why it doesn't make sense.

**Reply by helmis123** [reply]
@AbdelilahDerfoufi Please post a reproducer to help u.

**Reply by AbdelilahDerfoufi** [reply]
I don't know how

**Reply by chyshyya** [reply]
With php 8.0.8 everything works like expected with your files
```
	/**
	 * @Route("/test", name="app_test")
	 */
	public function test(): Response
	{
		$user = $this->getUser();
		if ($user) {
			$interventions = $user->getInterventions();
			dd($interventions->first());
		}
		
		return new Response("bad");
	}
```
![image](https://user-images.githubusercontent.com/994758/152824914-542d7104-4b46-4854-880c-85f1f7096471.png)

**Reply by AbdelilahDerfoufi** [reply]
Then maybe just visuel studio code gone crazy. 
I added first method and it worked for me as well.
I apologize for the troubles I have caused you

**Reply by chyshyya** [reply]
no problem. And my advise to use PhpStorm. It's better in all cases, especially for Symfony 2 and above. Have good coding.

**Reply by AbdelilahDerfoufi** [reply]
I was using but I am no longer  in the trial period. And there is no community lisence

**Reply by VincentTrotot** [reply]
I know the issue is answered but I think I understand why VSCode is acting like that, and I got a solution.

`$this->getUser()` type is `UserInterface`, so VSCode don't know which class implements the interface.  

You can help it with annotations : 

```php
/** @var $user User */
$user = $this->getUser();
```

That way VSCode know that `$user` type is `User`, therefore it knows its methods.

**Reply by AbdelilahDerfoufi** [reply]
Thank you for the information.

**Reply by Lyonict** [reply]
> I know the issue is answered but I think I understand why VSCode is acting like that, anf I got a solution.
> 
> `$this->getUser()` type is `UserInterface`, so VSCode don't know which class implements the interface.
> 
> You can help it with annotations :
> 
> ```
> /** @var $user User */
> $user = $this->getUser();
> ```
> 
> That way VSCode know that `$user` type is `User`, therefore it knows its methods.

Doing a bit of necromancy here, but since I faced the same issue I thought I'd contribute

Your comment put me on the right track, but in my case it was 
```
/** @var $user User */
$user = $this->getUser();
```
